oss-internship-2020/gdal/raster.cc

```diff
@@ -0,0 +1,144 @@
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <syscall.h>
+
+#include <iostream>
+
+#include "gdal_sapi.sapi.h"
+#include "sandboxed_api/sandbox2/util/fileops.h"
+
+class GdalSapiSandbox : public gdalSandbox {
+ public:
+  std::string filePath;
+
+  GdalSapiSandbox(std::string path)
+      : gdalSandbox(), filePath(std::move(path)) {}
+
+  std::unique_ptr<sandbox2::Policy> ModifyPolicy(
+      sandbox2::PolicyBuilder*) override {
+    return sandbox2::PolicyBuilder()
+        .AllowDynamicStartup()
+        .AllowRead()
+        .AllowSystemMalloc()
+        .AllowWrite()
+        .AllowExit()
+        .AllowStat()
+        .AllowOpen()
+        .AllowSyscalls({
+            __NR_futex,
+            __NR_close,
+            __NR_recvmsg,
+            __NR_getdents64,
+            __NR_lseek,
+            __NR_getpid,
+            __NR_sysinfo,
+            __NR_prlimit64,
+            __NR_ftruncate,
+            __NR_unlink,
+        })
+        .AddDirectory("/usr/local/lib")
+        .AddFile(filePath)
+        .BuildOrDie();
+  }
+};
+
+int main() {
+  // Reading GDALDataset from a (local, specific) file.
+  std::string filename =
+      "/usr/local/google/home/inach/work/sandboxed-api/oss-internship-2020/"
+      "gdal/CANYrelief1-geo.tif";
+
+  GdalSapiSandbox sandbox(filename);
+  sandbox.Init().IgnoreError();
```

Comment:
Can you please rework this file to use the same error-handling as your PFFFT sandbox?
I.e. like this:
```
SAPI_RETURN_IF_ERROR(sandbox.Init());
```

Using a `main()` function like
```c++
int main(int argc, char* argv[]) {
  // Optionally handle command line arguments here
  if (absl::Status status = GdalMain(); !status.ok()) {
    LOG(ERROR) << "Initialization failed: " << status.ToString();
    return EXIT_FAILURE;
  }

  return EXIT_SUCCESS;
}
```